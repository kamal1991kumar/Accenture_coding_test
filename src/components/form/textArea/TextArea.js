import React from "react";
import PropTypes from "prop-types";
export const TextArea = ({ hasError, message, value, handleChange }) => (
  <div className="textArea">
    <textarea
      defaultValue={value}
      data-testid="textArea"
      className={`textArea__input textArea__input-${hasError ? "error" : ""}`}
      onChange={({ currentTarget }) => handleChange(currentTarget.value)}
    />
    {hasError ? (
      <div data-testid="error" className="textArea__error">
        {message}
      </div>
    ) : null}
  </div>
);

TextArea.propTypes = {
  hasError: PropTypes.bool.isRequired,
  message: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired,
};
TextArea.defaultProps = {
  value: "",
};
