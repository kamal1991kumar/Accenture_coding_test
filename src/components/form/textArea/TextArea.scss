.textArea {
  &__input {
    width: 100%;
    border:solid 1px #ebebeb;
    padding: 8px 10px;
    font-size: 10px;
    line-height: 15px;
    max-width: 100%;
    min-width: 100%;
    height: 80px;
    &-error {
      border-color: red;
    }
  }
  &__error {
    color: red;
    font-size: 8px;
    padding: 3px;
  }
}