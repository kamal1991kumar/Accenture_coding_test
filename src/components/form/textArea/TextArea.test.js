import React from "react";
import { render, fireEvent } from "@testing-library/react";
import TextArea from "./index";

const props = {
  hasError: false,
  message: "textArea not be empty",
  value: "text",
  handleChange: jest.fn(),
};

describe("TextArea Component", () => {
  test("Should render on screen", () => {
    const { container } = render(<TextArea {...props} />);
    expect(container).toMatchSnapshot();
    expect(container).toBeInTheDocument();
  });

  test("Should render props value", () => {
    const { getByTestId } = render(<TextArea {...props} />);
    expect(getByTestId("textArea")).toHaveValue(props.value);
  });

  test("Should render changed value", () => {
    const { getByTestId } = render(<TextArea {...props} />);
    const textArea = getByTestId("textArea");
    fireEvent.change(textArea, {
      target: { value: "123" },
    });
    expect(textArea).toHaveValue("123");
  });
  test("Should render error message", () => {
    const { getByTestId, container } = render(<TextArea {...props} hasError />);
    const textArea = getByTestId("error");
    expect(container).toMatchSnapshot();
    expect(textArea.textContent).toBe(props.message);
  });
});
