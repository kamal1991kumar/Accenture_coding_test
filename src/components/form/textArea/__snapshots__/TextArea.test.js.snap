// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea Component Should render error message 1`] = `
<div>
  <div
    class="textArea"
  >
    <textarea
      class="textArea__input textArea__input-error"
      data-testid="textArea"
    >
      text
    </textarea>
    <div
      class="textArea__error"
      data-testid="error"
    >
      textArea not be empty
    </div>
  </div>
</div>
`;

exports[`TextArea Component Should render on screen 1`] = `
<div>
  <div
    class="textArea"
  >
    <textarea
      class="textArea__input textArea__input-"
      data-testid="textArea"
    >
      text
    </textarea>
  </div>
</div>
`;
